File: bus/src/main/java/com/kiylx/bus/eventbus/ipc/binder/model/EventMessage.java
package com.kiylx.bus.eventbus.ipc.binder.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * 进程间缓存事件封装类，事件定义
 */
public class EventMessage implements Parcelable {

   public String dataFrom;
   public String dataTo;
   public String channelName;
   public String connectService;
   public String dataType;
   public String json;


    public EventMessage() {
    }

    public EventMessage(final String channelName,
                        final String type) {
        this(channelName,  type, null);
    }

    public EventMessage(final String channelName,
                        final String type,
                        final String json
    ) {
        this.channelName = channelName;
        this.type = type;
        this.json = json;
    }

    protected EventMessage(Parcel in) {
        channelName = in.readString();
        type = in.readString();
        json = in.readString();
    }

    @Override
    public String toString() {
        return "{" +
                ", channel='" + channelName + '\'' +
                ", type='" + type + '\'' +
                ", json='" + json + '\'' +
                '}';
    }

    public static final Creator<EventMessage> CREATOR = new Creator<EventMessage>() {
        @Override
        public EventMessage createFromParcel(Parcel in) {
            return new EventMessage(in);
        }

        @Override
        public EventMessage[] newArray(int size) {
            return new EventMessage[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest, final int flags) {
        dest.writeString(channelName);
        dest.writeString(type);
        dest.writeString(json);
    }

    public void readFromParcel(Parcel reply) {
        channelName = reply.readString();
        type = reply.readString();
        json = reply.readString();
    }
}

File: bus/src/main/java/com/kiylx/bus/eventbus/ipc/binder/model/EventMessage.kt
package com.kiylx.bus.eventbus.ipc.binder.model

import android.os.Parcel
import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * 进程间缓存事件封装类，事件定义
 *@param dataFrom 来自哪个client进程
 *@param dataTo 去往哪个host进程
 *@param channelName 要监听的channel名称
 *@param connectService 连接到的服务。 service与host进程不是同一个进程。所以有dataTo和connectService的区别
 *@param dataType
 */
@Parcelize
data class EventMessage(
    val dataFrom: String="",
    val dataTo: String="",
    val channelName: String="",
    val connectService: String="",
    val dataType: String="",
    val json: String =""
) : Parcelable {
    fun readFromParcel(_reply: Parcel) {

    }
}


